pkgs/development/python-modules/pyzstd/default.nix

```diff
@@ -0,0 +1,48 @@
+{
+  buildPythonPackage,
+  fetchFromGitHub,
+  lib,
+  pkgs,
+  pypaInstallHook,
+  setuptoolsBuildHook,
+}:
+buildPythonPackage rec {
+  pname = "pyzstd";
+  version = "0.16.2";
+
+  # There is a pyproject.toml, but we want to dynamically link zstd, which can
+  # only be done through a setup.py argument
+  pyproject = false;
+
+  src = fetchFromGitHub {
+    owner = "Rogdham";
+    repo = "pyzstd";
+    tag = version;
+    hash = "sha256-Az+0m1XUFxExBZK8bcjK54Zt2d5ZlAKRMZRdr7rPcss=";
+  };
+
+  nativeBuildInputs = [
+    pypaInstallHook
+    setuptoolsBuildHook
+  ];
+
+  buildInputs = [
+    # zstd from top-level scope, not python-modules scope
+    pkgs.zstd
+  ];
+
+  setupPyBuildFlags = [
+    "--dynamic-link-zstd"
+  ];
+
```

Comment:
Please enable tests with `pythonImportsCheck` and `pytestCheckHook`.
https://nixos.org/manual/nixpkgs/unstable/#testing-python-packages